lang/src/org/partiql/lang/eval/visitors/SelectStarVisitorTransform.kt

```diff
@@ -0,0 +1,117 @@
+package org.partiql.lang.eval.visitors
+
+import com.amazon.ionelement.api.emptyMetaContainer
+import org.partiql.lang.ast.UniqueNameMeta
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.eval.errNoContext
+
+class SelectStarVisitorTransform : PartiqlAst.VisitorTransform() {
+
+    /**
+     * Copies all parts of [PartiqlAst.Expr.Select] except [newProjection] for [PartiqlAst.Projection].
+     */
+    private fun copyProjectionToSelect(node: PartiqlAst.Expr.Select, newProjection: PartiqlAst.Projection): PartiqlAst.Expr {
+        return PartiqlAst.build {
+            select(
+                setq = node.setq,
+                project = newProjection,
+                from = node.from,
+                fromLet = node.fromLet,
+                where = node.where,
+                group = node.group,
+                having = node.having,
+                limit = node.limit,
+                metas = node.metas)
+        }
+    }
```

Comment:
Again, not sure if there's a better way to make a copy of a PIG type with one parameter changed.

Reply:
For the time being, this is a reasonable work-around.  Please add a link to https://github.com/partiql/partiql-ir-generator/issues/52 on line 14 to remind us to change this to use the generated `.copy` function.

Reply:
Added.